adapters/lora-vertx-base/src/main/java/org/eclipse/hono/adapter/lora/providers/KerlinkProvider.java

```diff
@@ -61,33 +82,96 @@ public LoraMessageType getMessageType(final JsonObject loraMessage) {
         return LoraMessageType.UPLINK;
     }
 
-    /**
-     * {@inheritDoc}
-     * <p>
-     * This method always returns {@value #HEADER_CONTENT_TYPE_KERLINK_JSON}.
-     */
     @Override
-    public String acceptedContentType() {
-        return HEADER_CONTENT_TYPE_KERLINK_JSON;
+    protected String getDevEui(final JsonObject loraMessage) {
+
+        Objects.requireNonNull(loraMessage);
+        return LoraUtils.getChildObject(loraMessage, FIELD_KERLINK_END_DEVICE, JsonObject.class)
+            .map(endDevice -> endDevice.getValue(FIELD_KERLINK_DEVICE_EUI))
+            .filter(String.class::isInstance)
+            .map(String.class::cast)
+            .orElseThrow(() -> new LoraProviderMalformedPayloadException("message does not contain String valued device ID property"));
     }
 
     @Override
-    protected String getDevEui(final JsonObject loraMessage) {
+    protected Buffer getPayload(final JsonObject loraMessage) {
 
         Objects.requireNonNull(loraMessage);
-        return LoraUtils.getChildObject(loraMessage, FIELD_UPLINK_DEVICE_EUI, String.class)
-                .orElseThrow(() -> new LoraProviderMalformedPayloadException("message does not contain String valued device ID property"));
+        final String encodingType = LoraUtils.getChildObject(loraMessage, FIELD_KERLINK_ENCODING_TYPE, String.class).orElse("");
+        final Optional<String> payload = LoraUtils.getChildObject(loraMessage, FIELD_KERLINK_PAYLOAD, String.class);
+
+        switch (encodingType) {
+            default:
```

Comment:
to me it looks like the default is Base64 ...

Reply:
yes, you're right. adapted it